### ISSUE TITLE
Jellyfin Media Player Flatpak is unable to access (non local?) Jellyfin servers

### ISSUE THREAD

**Issue by Constancies** [flathub/com.github.iwalton3.jellyfin-media-player#47]
Whenever I attempt to access my server or Jellyfin's demo server, the Jellyfin Media Player client is unable to connect, with the error:
```We're unable to connect to the selected server right now. Please ensure it is running and try again.```
I'm not sure if it is able to access local servers either, as I haven't been able to test this just yet, but I do know that the Jellyfin Media Player `.deb` is perfectly capable of accessing non-local servers. Maybe this is a flatpak permission issue?

To reproduce, give this a shot with Jellyfin's demo instance, https://demo.jellyfin.org.

**Comment by NAHFE** [comment]
I seem to have the same problem on openSUSE Tumbleweed.
When looking through the logs there is this message
`2024-05-17 00:45:19.388 [debug] onCertificateError @ 221 - https://demo.jellyfin.org/system/info/public :Server's certificate is not trusted.-202`

**Comment by FranckRJ** [comment]
Same issue on openSUSE Tumbleweed, I also opened an issue on the JMP repo: [667](https://github.com/jellyfin/jellyfin-media-player/issues/667).

> I have another laptop running Linux Mint that doesn't have this bug

Which version of flatpak do you have in linux mint ? On Tumbleweed I have Flatpak 1.15.8.

**Comment by flo-at** [comment]
Same issue on ArchLinux with Flatpak 1.15.8. I tried JMP 1.9.1 and 1.10.1.
From the log:
```
  2024-05-31 23:23:06.010 [info] unknown @ 0 - JS: begin connectToServer
  2024-05-31 23:23:06.011 [info] unknown @ 0 - JS: tryReconnect: https://xxxxx
  2024-05-31 23:23:06.112 [info] unknown @ 0 - JS: getTryConnectPromise https://xxxxx
  2024-05-31 23:23:06.113 [info] unknown @ 0 - JS: ConnectionManager requesting url: https://xxxxx/system/info/public
  2024-05-31 23:23:06.114 [info] unknown @ 0 - JS: fetchWithTimeout: timeoutMs: 20000, url: https://xxxxx/system/info/public
  2024-05-31 23:23:06.126 [info] unknown @ 0 - JS: fetchWithTimeout: timed out connecting to url: https://xxxxx/system/info/public
  2024-05-31 23:23:06.126 [info] unknown @ 0 - JS: ConnectionManager request failed to url: https://xxxxx/system/info/public
  2024-05-31 23:23:06.127 [info] unknown @ 0 - JS: Reconnect failed to https://xxxxx
  2024-05-31 23:23:06.127 [info] unknown @ 0 - JS: connectToAddress https://xxxxx failed
```
The connection "timed out" immediately.

**Comment by FranckRJ** [comment]
> I tried JMP 1.9.1 and 1.10.1.

It's strange because for me JMP 1.9.1 worked but not 1.10.1. And my error message is not the same, I don't have a timeout but a `Server's certificate is not trusted` (mentioned earlier). Maybe we don't have the same issue, I don't know.

**Comment by flo-at** [comment]
It's probably something else, you're right. My problem is more general and most likely not related to Jellyfin directly but to Chromium which is used in Qt WebEngine.

**Comment by SuspiciousDuck** [comment]
Any update on this? Still experiencing the untrusted certificate error on flatpak jellyfin-media-player 1.12.0 with the server's jellyfin version as 10.10.7

**Comment by Aeris1One** [comment]
Do you use a self-signed certificate on your server ?

**Comment by SuspiciousDuck** [comment]
No, I use caddy, which gets a certificate from Let's Encrypt.

**Comment by Aeris1One** [comment]
1. Can you `curl` into the website ? I'm really talking about curl, not a browser access.

I'm asking because LetsEncrypt did recently add a new intermediate CA - R12 - to production, which, while is signed by the root, might not be directly trusted on every distro right now. I see demo.jellyfin.com cert is coming from that R12 CA.
Browsers often have their own chains of trusts which are better kept up to date than system ones, so don't use a browser to test, use `curl` or `wget`.

2. Can you try with the "Auto-detect certificate bundle" checked in client settings (might need to connect once without HTTPS to be able to check the setting) ?

And this one is in case the Jellyfin client doesn't trust LetsEncrypt R12, but your system does. It prompts Jellyfin Media Player to use the system's roots of trusts.

**Comment by SuspiciousDuck** [comment]
Yes, I can curl the site. I logged into the jellyfin server using http://192.168.0.5:8096/jelly, and when I checked the client settings, I saw that Auto-Detect Certificate Bundle was already checked.
